packages/asc-ui/src/components/Button/ButtonStyle.ts

```diff
@@ -1,41 +1,182 @@
-import styled, { css } from '@datapunt/asc-core'
-import { readableColor } from 'polished'
-import { color as getColor } from '../../utils'
-import ButtonBaseStyle, {
-  Props as ButtonBaseStyleProps,
-} from './ButtonBaseStyle'
+import styled, { Theme, css } from '@datapunt/asc-core'
+import { transitions, readableColor, darken, size } from 'polished'
+import {
+  color as themeColor,
+  focusStyleOutline,
+  color as getColor,
+  svgFill,
+} from '../../utils'
+import { flexboxMinHeightFix } from '../shared/ie-fixes'
 import IconStyle from '../Icon/IconStyle'
 
-export type Props = ButtonBaseStyleProps
-
-const ButtonStyle = styled(ButtonBaseStyle)<Props>`
-  padding: 4px 10px 4px 10px;
-  min-height: 36px;
-  line-height: 1em;
-  color: ${({ color, theme }) =>
-    color
-      ? readableColor(getColor(color)({ theme }))
-      : getColor('primary')({ theme })};
-  ${({ theme, color }) =>
-    !color &&
-    css`
-      border: 1px solid ${getColor('primary')({ theme })};
-    `}
-
-  &:hover {
-    ${({ theme, color }) =>
-      !color &&
-      css`
-        outline: 1px solid ${getColor('primary')({ theme })};
-      `}
-  }
+enum ButtonVariants {
+  primary,
+  secondary,
+  tertiary,
+  primaryInverted,
+  blank,
+}
+
+const getVariant = () => ({
+  theme,
+  variant,
+  color,
+}: {
+  theme: any
+  variant?: keyof typeof ButtonVariants
+  color?: Theme.TypeLevel
+}) => {
+  switch (variant) {
+    case 'primary':
+      return css`
+        background-color: ${themeColor('primary')({ theme })};
+        color: ${readableColor(themeColor('primary')({ theme }))};
+        ${svgFill('tint', 'level1')};
+
+        &:hover {
+          background-color: ${darken(0.1, themeColor('primary')({ theme }))};
+        }
+      `
+
+    case 'secondary':
+      return css`
+        background-color: ${themeColor('secondary')({ theme })};
+        color: ${themeColor('tint', 'level1')({ theme })};
+        ${svgFill('tint', 'level1')};
+
+        &:hover {
+          background-color: ${darken(0.1, themeColor('secondary')({ theme }))};
+        }
+      `
+
+    case 'tertiary':
+      return css`
+        background-color: ${themeColor('tint', 'level4')({ theme })};
 
-  & > ${/* sc-selector */ IconStyle}:first-child {
-    margin-right: 5px;
+        &:hover {
+          background-color: ${darken(
+            0.1,
+            themeColor('tint', 'level4')({ theme }),
+          )};
+        }
+      `
+
+    case 'primaryInverted':
+      return css`
+        color: ${getColor('primary')({ theme })};
+        border: 1px solid ${getColor('primary')({ theme })};
+        ${svgFill('primary')};
+
+        &:hover {
+          outline: 1px solid ${getColor('primary')({ theme })};
+        }
+      `
+
+    case 'blank':
```

Comment:
can pu please add a short description for this one? the rest of variants are clear wat they do.